tests/test_module.py

```diff
@@ -975,4 +988,37 @@ def test_rule_timebucket_64bit(self):
             r.execute_command("ts.createrule", 'test_key', 'above_32bit_limit', 'AGGREGATION', 'max', ABOVE_32BIT_LIMIT)
             info = r.execute_command("ts.info", 'test_key')
             assert info[19][0][1] == BELOW_32BIT_LIMIT            
-            assert info[19][1][1] == ABOVE_32BIT_LIMIT
\ No newline at end of file
+            assert info[19][1][1] == ABOVE_32BIT_LIMIT
+
+########## Test init args ##########
```

Comment:
Let's try to it thiser way:
```
def ModuleArgsTestCase(good, args):
    class _Class(ModuleTestCase(REDISTIMESERIES, module_args=args)):
        def test(self):
            ping = False
            try:
                ping = self.cmd('PING')
            except Exception:
                delattr(self, '_server') # server died, do not try to kill it
            assert good and ping or not good and not ping

    return _Class

class TestRetentionGood(ModuleArgsTestCase(True, ['RETENTION_POLICY', '100'])):
    pass

class TestRetentionBad_1(ModuleArgsTestCase(False, ['RETENTION_POLICY', 'BAD'])):
    pass

class TestRetentionBad_2(ModuleArgsTestCase(False, ['RETENTION_POLICY', -1])):
    pass

class TestMaxSamplesGood(ModuleArgsTestCase(True, ['MAX_SAMPLE_PER_CHUNK', '100'])):
    pass

class TestMaxSamplesBad_1(ModuleArgsTestCase(False, ['MAX_SAMPLE_PER_CHUNK', 'BAD'])):
    pass

class TestMaxSamplesBad_2(ModuleArgsTestCase(False, ['MAX_SAMPLE_PER_CHUNK', -1])):
    pass

class TestPolicyGood(ModuleArgsTestCase(True, ['COMPACTION_POLICY', 'max:1m:1d;min:10s:1h;avg:2h:10d;avg:3d:100d'])):
    pass

class TestPolicyBad(ModuleArgsTestCase(False, ['COMPACTION_POLICY', 'BAD'])):
    pass
```
Also removing ```assertInitArgsFail``` and ```assertInitArgsSuccess``` from the top.

Reply:
Very nice